### PR TITLE
Notebook

### PR DESCRIPTION
**Pull request by ocefpaf**
@emiliom this is the first step to add a few notebook examples here that we ca run with [binder](http://mybinder.org/) and the [ODM2](http://anaconda.org/odm2) channel.

### CODE REVIEW

File: notebooks/odm2api_sample_fromsqlite.ipynb [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Create a connection to the ODM2 database\n",
+    "# ----------------------------------------\n",
+    "odm2db_fpth = 'ODM2-LittleBear1.sqlite'\n",
```
Comment (ocefpaf) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: This is the file I mentioned on the call today that we need to run this notebook. Maybe we can create another version that does not need this file and/or use online resources if you cannot add the file here.

**Comment by emiliom** [comment]
See the email I just sent you. And sorry, but I have to go "offline" now and for probably the rest of the day. I need to work on an ODM2 "use cases" paper that's due today!! Then I have other calls and meetings.

I can follow up tomorrow if you still need something from me. FYI, I'm hoping to be able to test the new mocsy package tomorrow, too.

File: notebooks/odm2api_sample_fromsqlite.ipynb (hunk 1 above)
Comment (emiliom) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: _I'm pasting here the email I sent you on Friday:_

The odm2api demo notebook is here:
https://github.com/emiliom/stuff/blob/master/odm2api_sample_fromsqlite.ipynb

But note that it uses a SQLite file from my local drive, with a hardwired file path. I've uploaded that SQLite file (3.3 MB) into a repo of mine here:
https://github.com/emiliom/stuff/blob/master/ODM2-LittleBear1.sqlite

Feel free to grab both of those files and do whatever you need to do to create a binder live notebook demo. That'll be fun to see, thanks for offering to do that!

Comment (emiliom) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: I'll ping @horsburgh here so he's aware that:
- I've posted a tweaked version of his Little Bear SQLite ODM2 database
- The tweak I made was to create (manually) the SamplingFeatures.FeatureGeometryWKT field, and populate it based on the WKT in SamplingFeatures.FeatureGeometry (which, if I remember correctly, is still in the original, **incorrectly reversed** YX order)

Comment (horsburgh) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: @emiliom - I fixed both of these on Friday in my testing. Where did you post the sample Little Bear SQLite ODM2 database?

Comment (emiliom) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: @horsburgh - great. I posted my version [on a personal repo here](https://github.com/emiliom/stuff/blob/master/ODM2-LittleBear1.sqlite). But I can delete it. Can you upload yours somewhere more official (say, in your use case folder, https://github.com/ODM2/ODM2/tree/master/usecases/littlebearriver)? Please post the url here for @ocefpaf. Once you do that, I'll delete mine.

Comment (ocefpaf) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: > Please post the url here for @ocefpaf. Once you do that, I'll delete mine.

Sure. Right now I add your version in a branch just for testing, but I was not successful:

http://nbviewer.jupyter.org/github/ocefpaf/conda-recipes-ODM2/blob/notebook/notebooks/odm2api_sample_fromsqlite.ipynb

Comment (emiliom) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: You're using an older version of my notebook, before the changes I made to make it work with odm2api 0.5.0:
https://github.com/emiliom/stuff/blob/master/odm2api_sample_fromsqlite.ipynb

You may have been to fast ... I think you posted this PR before I had updated the notebook on my repo.

Comment (ocefpaf) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: @emiliom thanks for the new notebook. It is working. I am building the binder now.

Comment (emiliom) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: Great! Please send me an email when the binder is done and working; but no rush at all.

Comment (ocefpaf) [inline notebooks/odm2api_sample_fromsqlite.ipynb]: It is already :smile: 

Just click on the binder badge at the main page:

https://github.com/ODM2/conda-recipes-ODM2

**Comment by emiliom** [comment]
Awesome!! Also, nice reorganization of the repo ;)

While you're at it, I'll ask you to make an easy (for you) enhancement to the notebook: please add a Folium map with a marker for the site location. See cell 13, which returns "POINT (41.718473 -111.946402)". FYI, this Well Known Text (WKT) is in the wrong order for now (my fault); the first value is latitude, second is longitude. Please have them marker include a popup with a bit of HTML, based on string attributes at the site. Specifically:

```
sf.SamplingFeatureCode
sf.SamplingFeatureTypeCV
sf.SamplingFeatureName
```

You can add the Folium map right after cell 13. Thanks!

**Comment by horsburgh** [comment]
@emiliom and @ocefpaf - I just uploaded a fixed version of the ODM2.sqlite example in this folder:

https://github.com/ODM2/ODM2/tree/master/usecases/WOF_to_ODM2

It should have the FeatureGeometryWKT field populated and the latitude and longitude points should be in the right order.

**Comment by ocefpaf** [comment]
> While you're at it, I'll ask you to make an easy (for you) enhancement to the notebook: please add a Folium map with a marker for the site location.

See https://github.com/ODM2/conda-recipes-ODM2/pull/37

> See cell 13, which returns "POINT (41.718473 -111.946402)". FYI, this Well Known Text (WKT) is in the wrong order for now (my fault); the first value is latitude, second is longitude. Please have them marker include a popup with a bit of HTML, based on string attributes at the site. Specifically:

It seems that asking for `lon, lat = sf.shape().x, sf.shape().y` returns the correct values.

> @emiliom and @ocefpaf - I just uploaded a fixed version of the ODM2.sqlite example in this folder

Thanks @horsburgh!

PS: Should we convert those examples to notebooks and add them here?

**Comment by emiliom** [comment]
>  asking for lon, lat = sf.shape().x, sf.shape().y returns the correct values.

Thanks for being adventurous and trying out the shape() methods! (I wrote them, so it's heart-warming). Are you now using @horsburgh's SQLite file rather than mine?

The Folium map looks great, thanks! Hopefully @horsburgh will be pleased to see his Little Bear site on a dynamic embedded map on a notebook :smiley: 

> PS: Should we convert those examples to notebooks and add them here?

What examples? If you're referring to @horsburgh's comment, he was referring to the SQLite database file (it's a file we're using heavily as an example/demo ODM2 database).

**Comment by ocefpaf** [comment]
> Are you now using @horsburgh's SQLite file rather than mine?

Yours. I did not update the file.

> If you're referring to @horsburgh's comment, he was referring to the SQLite database file (it's a file we're using heavily as an example/demo ODM2 database).

@horsburgh has [load_wof_to_ODM2.py](https://github.com/ODM2/ODM2/blob/master/usecases/WOF_to_ODM2/load_wof_to_ODM2.py) and [create_ODM2_sqlite.py](https://github.com/ODM2/ODM2/blob/master/usecases/WOF_to_ODM2/create_ODM2_sqlite.py) there that could be added here as examples if you think it is worth it.

Edit: I am just browsing for more examples to add to the binder.

**Comment by emiliom** [comment]
> Yours. I did not update the file.

Please update it when you get a chance. It's safer to be using the real deal, rather than my copy.

> @horsburgh has load_wof_to_ODM2.py and create_ODM2_sqlite.py there that could be added here as examples if you think it is worth it.
> 
> Edit: I am just browsing for more examples to add to the binder.

Ah, ok. 

@horsburgh can decide if those two are worth turning into notebooks at this time. My only suggestion is to move forward with that (if that's the decision) using a new, separate issue! We've abused the comments on this PR too much already :grimacing:  ... It's getting confusing.

Thanks again for getting that binder out, and adding the Folium elements!!

**Comment by ocefpaf** [comment]
> We've abused the comments on this PR too much already :grimacing: ... It's getting confusing.

:smile: 

> Please update it when you get a chance. It's safer to be using the real deal, rather than my copy.

That will lead to some changes in the example. Mostly remove the failing cells. (See http://nbviewer.jupyter.org/gist/ocefpaf/b38f6612f9564e0e01a24239c8305b45)

**Comment by emiliom** [comment]
> That will lead to some changes in the example

Yeah, I'm not surprised. BTW, thanks for the nice cleanups (better syntax) you've already done. Also, I see that `sf.shape()` no longer works (I know why -- no worries); but it's also nice to see the following working:

``` python
lon, lat = sf.Longitude, sf.Latitude
```

That's a new feature in odm2api 0.5.0 .... Glad you discovered it.

**Pull request merged**